File: 566-reshape-the-matrix/566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        m, n = len(mat), len(mat[0])
        if m * n != r * c:
            return mat
        
        nmat = [[0] * c for _ in range(r)]
        
        i = j = 0
        for row in mat:
            for col in row:
                nmat[i][j] = col
                if j == c - 1:
                    i += 1
                    j = 0
                else:
                    j += 1
        return nmat

File: 566-reshape-the-matrix/566-reshape-the-matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        int m = mat.length, n = mat[0].length;
        if(r * c != m * n)
            return mat;
        
        int[][] ans = new int[r][c];
        int i = 0, j = 0;
        for(int[] row : mat){
            for (int col : row){
                ans[i][j] = col;
                j++;
                
                if(j == c){
                    j = 0;
                    i++;
                }
            }
        }
        return ans;
    }
}